sklego/pandas_utils.py

```diff
@@ -0,0 +1,18 @@
+import inspect
+from functools import wraps
+import datetime as dt
+import logging
+
+
+def log_step(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        logger = logging.getLogger(inspect.stack()[1].function)
```

Comment:
maybe a comment that explains what `inspect.stack()` is for. it doesn't feel totally obvious. @MBrouns it's been suggested that you are involved in this discussion. @dominiquemaria correct?

Reply:
I usually put my `log_step` in a separate `utils.py` module but prefer the logs to not all be coming from this utils module, that's why I use inspect to get the `__name__` one stack up the stackframe

Reply:
@dominiquemaria feel free not to test for all circumstances initially. ill gladly merge this work if you take care of the other two items.

Reply:
Fair enough! agreed